azure-pipelines/build.yml

```diff
@@ -1,26 +1,63 @@
 parameters:
-- name: prereleaseFlag
+- name: versionNumberOverride
   type: string
+  default: 'default'
 
-steps:
-- checkout: self
-  clean: true
-  submodules: true
-  fetchTags: false
-  fetchDepth: 0
+stages:
+- stage: Build
+  displayName: 'Build VSIXs'
+  jobs:
+  - job:
+    displayName: 'Build Prerelease VSIXs'
+    steps:
+    - checkout: self
+      clean: true
+      submodules: true
+      fetchTags: false
+      fetchDepth: 0
 
-- template: prereqs.yml
+    - template: prereqs.yml
+      parameters:
+        versionNumberOverride: ${{ parameters.versionNumberOverride }}
 
-- script: gulp 'vsix:release:package' ${{ parameters.prereleaseFlag }}
-  displayName: 'Build VSIXs'
+    - script: gulp 'vsix:release:package' --prerelease
+      displayName: 'Build VSIXs'
+
+    - task: PublishPipelineArtifact@1
+      # Run the publish step so we have vsix's even if the tests fail.
+      condition: succeededOrFailed()
+      displayName: 'Publish VSIXs'
+      inputs:
+        targetPath: '$(Build.SourcesDirectory)/vsix'
+        artifactName: 'VSIX_Prerelease_$(System.JobAttempt)'
+
+    - script: npm run omnisharptest:artifacts
+      displayName: 'Run artifacts tests'
+
+  - job:
+    displayName: 'Build Release VSIXs'
+    steps:
+    - checkout: self
+      clean: true
+      submodules: true
+      fetchTags: false
+      fetchDepth: 0
+
+    - template: prereqs.yml
+      parameters:
+        versionNumberOverride: ${{ parameters.versionNumberOverride }}
+
+    - script: gulp 'vsix:release:package'
+      displayName: 'Build VSIXs'
 
-- task: PublishPipelineArtifact@1
-  # Run the publish step so we have vsix's even if the tests fail.
-  condition: succeededOrFailed()
-  displayName: 'Publish VSIXs'
-  inputs:
-    targetPath: '$(Build.SourcesDirectory)/vsix'
-    artifactName: 'VSIXs - Attempt $(System.JobAttempt)'
+    - task: PublishPipelineArtifact@1
+      # Run the publish step so we have vsix's even if the tests fail.
+      condition: succeededOrFailed()
+      displayName: 'Publish VSIXs'
+      inputs:
+        targetPath: '$(Build.SourcesDirectory)/vsix'
```

Comment:
can we make the target paths be different for prerelease and release packet of vsix'es? Might preempt dogfooding and testing confusions.

Reply:
Is this what you mean?  THey do get uploaded to different paths
![image](https://github.com/dotnet/vscode-csharp/assets/5749229/8deeace8-186b-41cb-9c96-15da91597d1d)

Reply:
Yes